### ISSUE TITLE
finding the "right" includes and library

### ISSUE THREAD

**Issue by ThomasWaldmann** [borgbackup/borg#1045]
i just played with libsodium and noticed an issue:

if one has 2 libsodium installed (1.0.8 from ubuntu in /usr/{include,lib} and 1.0.10 manually installed into /usr/local/{include,lib}, i think it should prefer the manually installed stuff in /usr/local (but it currently picks the old one from /usr).

so, i guess include_dirs and library_dirs need to get reordered, so it looks into /usr last.

**Comment by ThomasWaldmann** [comment]
just do that or is there a better way?:
```
for dir, path in [('/usr/include', include_dirs), ('/usr/lib', library_dirs), ]:
    if dir in path:
        path.remove(dir)
        path.append(dir)
```

**Comment by FelixSchwarz** [comment]
I'm hardly qualified to comment about borg but I think you should not dynamically add library directories. For example the user might want to use borg and develop one of the required libraries (e.g. libsodium). Having a locally installed version interfere with a system-installed program is highly confusing/surprising.

For example in Fedora and EPEL we try to replace all "#!/usr/bin/env python" with a hard-coded executable path like "#!/usr/bin/python" to ensure that no active virtualenv will mess up the program execution.

Personally I use virtualenvs to add local dev versions. Another acceptable option would be to add directories explicitly configured by the administrator.
(Sorry if I misunderstood your question, please disregard then.)

**Comment by ThomasWaldmann** [comment]
Just to be clear: I was talking about the code in borgbackup's setup.py that is used when installing borg from source or from a python package.

**Comment by ThomasWaldmann** [comment]
hmm, how does pkg-config (see #1925) deal with multiple versions of same lib installed in different locations?

**Comment by ThomasWaldmann** [comment]
Guess the search order can be solved using PKG_CONFIG_PATH in #1925.
